src/System.Private.Interop/src/InteropExtensions/PInvokeMarshal.Unix.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Security;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// This PInvokeMarshal class should provide full public Marshal 
+    /// implementation for all things related to P/Invoke marshalling
+    /// </summary>
+    public partial class PInvokeMarshal
+    {
+        public static void SaveLastWin32Error()
+        {
+            s_lastWin32Error = Interop.Sys.GetErrNo();
```

Comment:
This will call into the internal CoreCLR shims. Mono does not have those; and they are not guaranteed to be stable or present on CoreCLR.

Instead, you want to forward these to CoreCLR's System.Runtime.Marshal

Reply:
Thanks for catching this. Moving SaveLastWin32Error into PInvokeMarshal.cs and it will call Marshal API